docs/src/test/java/docs/http/javadsl/CustomHeaderExampleTest.java

```diff
@@ -0,0 +1,118 @@
+package docs.http.javadsl;
+
+import akka.http.javadsl.model.HttpHeader;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.headers.ModeledCustomHeader;
+import akka.http.javadsl.model.headers.ModeledCustomHeaderCompanion;
+import akka.http.javadsl.model.headers.RawHeader;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.japi.JavaPartialFunction;
+import org.junit.Test;
+import scala.PartialFunction;
+
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+public class CustomHeaderExampleTest extends JUnitRouteTest {
+  //#modeled-api-key-custom-header
+  public static class ApiTokenHeader extends ModeledCustomHeader {
+
+    ApiTokenHeader(String name, String value) {
+      super(name, value);
+    }
+
+    public boolean renderInResponses() {
+      return false;
```

Comment:
Well this isn't about "fake headers", this is about if someone uses an "APiBlaBla" header in their apps and wants to expose it as a type in Akka HTTP apps.

The discusion about headers for carrying info we can indeed have in #911